src/main/java/com/google/monitoring/runtime/instrumentation/NullInstrumentation.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2009 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.monitoring.runtime.instrumentation;
+
+import java.lang.instrument.ClassDefinition;
+import java.lang.instrument.ClassFileTransformer;
+import java.lang.instrument.Instrumentation;
+import java.lang.instrument.UnmodifiableClassException;
+import java.util.jar.JarFile;
+
+/**
+ * Implementation of the {@code Instrumentation} interface that does nothing.
+ * Intended to be used in {@link AllocationRecorder} to work around a bug in certain JVMs
+ * where calling {@code Instrumentation.getObjectSize()} during JVM shutdown triggers a
+ * JVM-crashing assert.
+ */
+class NullInstrumentation implements Instrumentation {
+  @Override
+  public void addTransformer(final ClassFileTransformer transformer, final boolean canRetransform) {
+  }
```

Comment:
IWBN to throw a RuntimeException if anyone ever called any method of this class other than getObjectSize().